pyfar/classes/filter.py

```diff
@@ -434,33 +442,50 @@ def init_state(self, cshape, state='zeros'):
                 new_state[idx, ...] = spsignal.lfilter_zi(coeff[0], coeff[1])
         super().init_state(state=new_state)
 
+    def impulse_response_length(self):
+        """
+        Get the length of filter impulse response in samples.
+
+        The length is computed from the last non-zero coefficient per channel.
+
+        Returns
+        -------
+        impulse_response_length : array
+            An integer array of shape (C, ) containing the length in samples
+            where `C` denotes the number of channels of the filter.
```

Comment:
is it intended, that we dont use the ``cshape``concept here?

Reply:
According to the docstrings, filters can not have multidimensional channel shapes. Hence, I did not use cshape here.